### PR TITLE
[#10176] Fix Admin Search Issues from UAT

### PR DESCRIPTION
**Pull request by madanalogy**
Part of #10176 

### Original Filed Issues

- [X] "Search" button is expected to be blue (active), but is currently grey (inactive)
- [X] In a new keyword search, when no result is found, previous search result is still present. (In v6, no result is shown)
- [X] No header widget to indicate that single result entry can be expanded.
- [X] Click “Google ID”, “Manage this account”, “Reset Google ID” will expand the result entry before redirection (should stop event propagation)
- [X] In v6, all redirections (“Google ID”, “Manage this account”, “Reset Google ID”) are handled by opening in a new tab. In v7, redirection happens in the current page. When going back, no result is cached. (Suggest to follow v6 style)
- [X] Reset Google ID option should have a warning pop-up, like “Regenerate links” does. 
- [X] ~~When input ` . ` or `-` as the search input, no result is found in v6, but everything is returned in v7.~~ _Unable to replicate, possibly result of previous search_

View of dropdown button for visual representation of expansion:

Screenshot 2020-06-17 at 2 25 35 PM

### CODE REVIEW

**Comment by madanalogy** [comment]
The addition of a modal for google id reset has impacted the e2e testing for that page. I am remedying the problem now.

**Comment by madanalogy** [comment]
@jtankw3 @wkurniawan07 @xpdavid I'm not sure why the e2e tests are still failing, would appreciate any insight into the issue :")

File: src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java [hunk 1]
```diff
@@ -150,6 +153,9 @@ public void resetStudentGoogleId(StudentAttributes student) {
         WebElement studentRow = getStudentRow(student);
```
Comment (xpdavid) [inline src/e2e/java/teammates/e2e/pageobjects/AdminSearchPage.java:153]: The stack trace shows that `getStudentRow` fail to find element.

**Comment by madanalogy** [comment]
Credits to @jtankw3 for the suggestion of changing the xpath search format

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html [hunk 2]
```diff
@@ -36,16 +36,28 @@
...
         <ng-container *ngFor="let instructor of instructors; let i = index">
           <tr (click)="instructor.showLinks = !instructor.showLinks">
-            <td>{{ instructor.courseId }}</td> <!-- TODO: tooltip course name -->
+            <td>
+              {{ instructor.courseId }}
+              <br/>
+              <div class="col-sm-1">
+                <button *ngIf="!instructor.showLinks"
+                        class="btn"><i class="fas fa-chevron-circle-down"
+                                       style="color: blue;"></i></button>
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.html:45]: I normally don't pick on indentations, but this is very weird.
Why not just do a simple one:
```html
<button *ngIf="!instructor.showLinks" class="btn">
  <i class="fas fa-chevron-circle-down" style="color: blue;"></i>
</button>
```

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html [hunk 3]
```diff
@@ -36,16 +36,28 @@
...
+                <button *ngIf="instructor.showLinks"
+                        class="btn"><i class="fas fa-chevron-circle-up" style="color: blue;"></i></button>
+              </div>
+            </td> <!-- TODO: tooltip course name -->
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.html:49]: Why defer this TODO?

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html [hunk 4]
```diff
@@ -94,17 +106,30 @@
       <tbody>
         <ng-container *ngFor="let student of students; let i = index">
           <tr (click)="student.showLinks = !student.showLinks">
-            <td>{{ student.courseId }} [{{ student.section }}] ({{ student.team }})</td> <!-- TODO: tooltip course name -->
+            <td>
+              {{ student.courseId }} [{{ student.section }}] ({{ student.team }})
+              <br/>
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.html:111]: Self-closing HTMLs should not be closed.
```suggestion
              <br>
```

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html [hunk 5]
```diff
@@ -94,17 +106,30 @@
...
             <td>{{ student.comments }}</td>
             <td>
-              <a href="{{ student.manageAccountLink }}" *ngIf="student.googleId && student.manageAccountLink">
+              <a href="{{ student.manageAccountLink }}" *ngIf="student.googleId && student.manageAccountLink"
+                 (click)="$event.stopPropagation()" target="_blank" rel="noopener noreferrer">
                 <i class="fas fa-info-circle"></i> Manage this account
               </a>
               <br>
-              <a id="reset-student-id-{{i}}" href="javascript:;" (click)="resetStudentGoogleId(student, $event)" *ngIf="student.googleId">
+              <a id="reset-student-id-{{i}}" href="javascript:" (click)="resetStudentGoogleId(student, $event)" *ngIf="student.googleId">
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/admin-search-page.component.html:132]: Unrelated change?

File: src/web/app/pages-admin/admin-search-page/reset-google-id-confirm-modal/reset-google-id-confirm-modal.component.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<div class="modal-header bg-warning">
+    <h5 class="modal-title"><i class="fas fa-exclamation-circle"></i> Warning: This account's Google ID will be reset. </h5>
```
Comment (wkurniawan07) [inline src/web/app/pages-admin/admin-search-page/reset-google-id-confirm-modal/reset-google-id-confirm-modal.component.html:2]: 2-space indentation for HTML/SCSS/TS files

File: src/web/app/pages-admin/admin-search-page/admin-search-page.component.html (hunk 2 above)
Comment (madanalogy) [reply src/web/app/pages-admin/admin-search-page/admin-search-page.component.html:45]: This was the unfortunate victim of copy-paste :P
